Adding MTU tests in reference to projectcalico/calico #2200

insequent opened the pull request:
## Description
This is a simple testing change that specifically checks custom MTU values. It was created in an attempt to replicate https://github.com/projectcalico/calico/issues/2200

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/projectcalico/cni-plugin?pullRequest=642) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/projectcalico/cni-plugin?pullRequest=642) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/projectcalico/cni-plugin?pullRequest=642) it.</sub>

insequent commented:
Try as I might, I cannot get the git diff to not look crazy for `calico_cni_test.go`, despite there being minimal changes. Anyways, changes since last PR:

* Removed `Describe("Run Calico CNI plugin"` from all tests
* Rearranged `Context("deprecate Hostname for nodename"` tests (there were 2) to be together under the same context.
* Minor comment update to utils' DetermineNodeName. It seems priorities had changed there, but the comment hadn't. Side note: if it's a priority list, we should start with the highest priority and return if a value is found, instead of rewriting the same variable and checking everything? Trying not to let my scope creep too much in this PR...
